src/main/ngapp/accio/src/app/editor/editor.component.ts

```diff
@@ -740,3 +796,58 @@ interface CursorPos {
   x: number;
   y: number;
 }
+
+class Rectangle {
+  imageWidth: number;
+  imageHeight: number;
+  brushRadius: number;
+  // Lowest y touched by the brush.
+  top: number;
+  // Highest y touched by the brush. 
+  bottom: number;
+  // Lowest x touched by the brush.
+  left: number;
+  // Highest x touched by the brush.
+  right: number;
+  
+  constructor(imageWidth: number, imageHeight: number, brushWidth: number, coord: Coordinate) {
+    this.imageWidth = imageWidth;
+    this.imageHeight = imageHeight;
+    this.brushRadius = brushWidth / 2;
+
+    this.top = (coord.y - this.brushRadius) >= 0 ? (coord.y - this.brushRadius) : 0;
+    this.bottom = (coord.y + this.brushRadius) <= (this.imageHeight - 1) ? (coord.y + this.brushRadius) : (this.imageHeight - 1);
+    
+    this.left = (coord.x - this.brushRadius) >= 0 ? (coord.x - this.brushRadius) : 0;
+    this.right = (coord.x + this.brushRadius) <= (this.imageWidth - 1) ? (coord.x + this.brushRadius) : (this.imageWidth - 1);
+  }
+
+  /** Compares the index's left, right, top, and bottom most pixel based on the brush radius to the current max and mins of all members */
+  compare(coord: Coordinate) {
```

Comment:
Can we come up with a more descriptive name?

When it was used above

```
 this.searchRectangle.compare(pixel);
```

I didn't know what was happening with function call and thought "what is it comparing? should this be in a if statement?".